### PR TITLE
Use built-in LUIS functionality; use the local hero repository instead of calling API

### PR DESCRIPTION
**Pull request by joudoki**

### CODE REVIEW

File: Dialogs/HeroDialog.js [hunk 1]
```diff
@@ -1,78 +1,86 @@
...
+
+                const resolutions = heroEntity.resolution.values;
+
+                if (resolutions.length === 0) {
+                    session.send('I was unable to resolve that hero');
+                    return;
+                }
+
+                if (resolutions.length > 1) {
+                    session.send('I\m not sure what hero you are talking about');
```
Comment (mloewent) [inline Dialogs/HeroDialog.js]: ?

File: Cards/HeroCard/HeroCard.js [hunk 2]
```diff
@@ -15,23 +14,23 @@ class HeroCard {
         this.cardAttachment = HeroCardAttachment;
 
         // Populate card header content
-        this.cardAttachment['content']['body'][0]['items'][0]['columns'][0]['items'][0]['url'] = this.image;
+        this.cardAttachment['content']['body'][0]['items'][0]['columns'][0]['items'][0]['url'] = this.image || "http://localhost/missing.png";
```
Comment (mloewent) [inline Cards/HeroCard/HeroCard.js]: use a hosted image so this won't break when deployed

**Review by mloewent — CHANGES_REQUESTED** [review]
Change missing image to hosted.

File: Dialogs/HeroDialog.js (hunk 1 above)
Comment (joudoki) [reply Dialogs/HeroDialog.js]: Whoops! fixed.

**Review by mloewent — APPROVED** [review]

**Pull request merged**